libc-bottom-half/headers/public/wasix/call_dynamic.h

```diff
@@ -0,0 +1,14 @@
+#ifndef wasix_call_dynamic_h
+#define wasix_call_dynamic_h
+// Wrappers for the dynamic call interface.
+//
+// Currently they are just aliases for the raw syscall, but this may change in the future if we need to
+// add additional logic before or after the call is made.
+
+#include <wasi/api_wasi.h>
+#include <wasi/api_wasix.h>
+#include <wasix/function_pointer.h>
+
+#define wasix_call_dynamic __wasi_call_dynamic
```

Comment:
Putting these into a separate header is a good idea! However, I'd put all of them in one header, or add one that includes all the "call-related" functionality, namely this, the closures, the function pointer stuff, and value types. C headers are usually bundles of functionality, not just one or two definitions. It's easier to use that way too. I suggest `wasix/dyncall.h` or similar.